### PR TITLE
Dynamic information_schema configuration and port more tests

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/4495

# Rationale for this change

Currently the information schema can not be enabled after the `SessionState` is created and it can not be disabled.

This means I can't enable it programmatically in the sqllogicttests, and makes for a strange user experience where you can run `set datafusion.catalog.information_schema = true` but it is a noop

# What changes are included in this PR?
1. Enable / Disable information schema dynamically
2. Port some `information_schema` tests that rely on this behavior 

# Are these changes tested?
Yes

# Are there any user-facing changes?

Users can now enable / disable the information schema

Example:

```sql
DataFusion CLI v15.0.0
❯ show tables;
+---------------+--------------------+-------------+------------+
| table_catalog | table_schema       | table_name  | table_type |
+---------------+--------------------+-------------+------------+
| datafusion    | information_schema | tables      | VIEW       |
| datafusion    | information_schema | views       | VIEW       |
| datafusion    | information_schema | columns     | VIEW       |
| datafusion    | information_schema | df_settings | VIEW       |
+---------------+--------------------+-------------+------------+
4 rows in set. Query took 0.045 seconds.
❯ set datafusion.catalog.information_schema = false;
0 rows in set. Query took 0.001 seconds.
❯ show tables;
Plan("SHOW TABLES is not supported unless information_schema is enabled")
```

### CODE REVIEW

File: datafusion/core/src/execution/context.rs [hunk 1]
```diff
@@ -1546,17 +1550,10 @@ impl SessionState {
...
-                Arc::new(CatalogWithInformationSchema::new(
-                    Arc::downgrade(&catalog_list),
-                    Arc::new(default_catalog),
-                ))
-            } else {
-                Arc::new(default_catalog)
-            };
-            catalog_list
-                .register_catalog(config.default_catalog.clone(), default_catalog);
+            catalog_list.register_catalog(
```
Comment (alamb) [inline datafusion/core/src/execution/context.rs:1553]: The InformationSchema changes are now handed in `update_information_schema`

File: datafusion/core/tests/sql/information_schema.rs [hunk 2]
```diff
@@ -30,91 +30,6 @@ use rstest::rstest;
 
 use super::*;
 
-#[tokio::test]
```
Comment (alamb) [inline datafusion/core/tests/sql/information_schema.rs:33]: I ported a few tests over to show the behavior -- I will port what else I can from this file as a follow on PR

File: datafusion/core/tests/sqllogictests/src/main.rs [hunk 3]
```diff
@@ -130,12 +130,6 @@ async fn context_for_test_file(file_name: &str) -> SessionContext {
             setup::register_aggregate_tables(&ctx).await;
             ctx
         }
-        "information_schema.slt" => {
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/src/main.rs:133]: since this is now controllable dynamically, we do not have to hard code the sqltest runner

**Comment by alamb** [comment]
FYI @tustvold

**Review by xudong963 — APPROVED** [review]
👍

**Comment by alamb** [comment]
cc @mvanschellebeeck

**Pull request merged**